userspace/libscap/test/scap_event.ut.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright (C) 2022 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+#include "scap.h"
+#include <gtest/gtest.h>
+#include "../../common/strlcpy.h"
+
+// This function behaves exactly like scap_event_encode_params but it will allocate the event and return it by setting the event pointer.
+int32_t scap_event_generate(scap_evt **event, char *error, enum ppm_event_type event_type, uint32_t n, ...)
```

Comment:
I would rename the `n` param in the signature in something more meaningful like `n_params` or `n_evt_args`...

Reply:
It's explained in the prototype for the `scap_event_encode..` function as it works in the same way. Should be ok to keep it like this I think.